mitodl/micromasters
# Exam Coupons: use edX coupons when user loads the dashboard

#### What are the relevant tickets?
Fix #4770 

#### What's this PR do?

- Adds a model for edx exam coupons, 
- modifies the management command `populate_edx_exam_coupons` to create instances of `ExamRunCoupon`.
- modifies the dashboard api to assign a coupon to a user if the user is authorized for an exam run when the user loads the dashboard.

#### How should this be manually tested?
Ask me for an example coupon file.
Run the management command `populate_edx_exam_coupons`.
Make sure your user is eligible to take an exam for a course
Load the dashboard, and make sure you have the exam link available to you.
